### PR TITLE
Simplify magic method definition code.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #88017
* #88030

It turns out sym_float (and the hypothetical sym_int) can
be defined in the same way as conventional magic methods.
Do so.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/__init__.py [hunk 1]
```diff
@@ -210,16 +210,16 @@ def __init__(self, node):
...
-    # Magic methods installed later
-
     def __bool__(self):
         return self.node.bool_()
 
     def __int__(self):
         return self.node.int_()
 
+    # Magic methods installed by torch.fx.experimental.symbolic_shapes
+
```
Comment (albanD) [inline torch/__init__.py:220]: Why leave an empty method here? Just for typing?

**Review by albanD — APPROVED** [review]

File: torch/__init__.py (hunk 1 above)
Comment (ezyang) [reply torch/__init__.py:220]: typing yup

File: torch/fx/experimental/symbolic_shapes.py [hunk 2]
```diff
@@ -258,22 +245,28 @@ def eval(cls, base, divisor):
...
+    raise NotImplementedError()
+
 magic_methods = {
     **reflectable_magic_methods,
     'eq': lambda a, b: sympy.Eq(a, b),
     'gt': lambda a, b: sympy.Gt(a, b),
     'lt': lambda a, b: sympy.Lt(a, b),
     'le': lambda a, b: sympy.Le(a, b),
     'ge': lambda a, b: sympy.Ge(a, b),
+    'sym_float': lambda a: a,  # TODO: why can't I wrap with sympy.Float?
```
Comment (anjali411) [inline torch/fx/experimental/symbolic_shapes.py:258]: not sure why this works ...

**Pull request closed without merging**